### PR TITLE
[Impeller] fix plus blend mode in porterduff shader.

### PR DESCRIPTION
**Pull request by jonahwilliams**
DrawAtlas already has the optimization in https://github.com/flutter/engine/pull/51778 and so it is still rendering incorrectly with wide gamut.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
flutter_01

**Comment by jonahwilliams** [comment]
After:

flutter_02


Note: I have no idea why the position changes some times, I think that is just a bug in the widget but 🤷‍♂️

File: impeller/entity/entity_unittests.cc [hunk 1]
```diff
@@ -1446,6 +1446,45 @@ TEST_P(EntityTest, DrawAtlasNoColor) {
   ASSERT_TRUE(OpenPlaygroundHere(std::move(e)));
 }
 
+TEST_P(EntityTest, F16WideGamutDrawAtlasPlus) {
```
Comment (gaaclarke) [inline impeller/entity/entity_unittests.cc:1449]: This isn't going to be wide gamut, the name of the test has to end with "WideGamut".  Can you add an assert that it is applied like the other ones please.

Comment (jonahwilliams) [reply impeller/entity/entity_unittests.cc]: Done

File: impeller/entity/entity_unittests.cc [hunk 2]
```diff
@@ -1446,6 +1446,48 @@ TEST_P(EntityTest, DrawAtlasNoColor) {
   ASSERT_TRUE(OpenPlaygroundHere(std::move(e)));
 }
 
+TEST_P(EntityTest, DrawAtlasPlusWideGamut) {
```
Comment (jonahwilliams) [inline impeller/entity/entity_unittests.cc:1449]: Did I need to opt this into a golden somewhere?

Comment (jonahwilliams) [reply impeller/entity/entity_unittests.cc]: Ahh duh

**Review by gaaclarke — APPROVED** [review]
lgtm!

**Pull request merged**